packages/provider/src/image-model/v1/image-model-v1.ts

```diff
@@ -0,0 +1,77 @@
+import { JSONValue } from '../../json-value/json-value';
+
+/**
+Image generation model specification version 1.
+ */
+export type ImageModelV1 = {
+  /**
+The image model must specify which image model interface
+version it implements. This will allow us to evolve the image
+model interface and retain backwards compatibility. The different
+implementation versions can be handled as a discriminated union
+on our side.
+   */
+  readonly specificationVersion: 'v1';
+
+  /**
+Name of the provider for logging purposes.
+   */
+  readonly provider: string;
+
+  /**
+Provider-specific model ID for logging purposes.
+   */
+  readonly modelId: string;
+
+  /**
+Generates an array of images.
+   */
+  doGenerate(options: {
+    /**
+Prompt for the image generation.
+     */
+    prompt: string;
+
+    /**
+Number of images to generate.
+     */
+    n: number;
+
+    /**
+Size of the images to generate. Must have the format `{width}x{height}`.
+     */
+    size: `${number}x${number}` | undefined;
+
+    /**
+Additional provider-specific options that are passed through to the provider
+as body parameters.
+
+The outer record is keyed by the provider name, and the inner
+record is keyed by the provider-specific metadata key.
+```ts
+{
+  "openai": {
+    "style": "vivid"
+  }
+}
+```
+     */
+    providerOptions: Record<string, Record<string, JSONValue>>;
+
+    /**
+Abort signal for cancelling the operation.
+     */
+    abortSignal?: AbortSignal;
+
+    /**
+  Additional HTTP headers to be sent with the request.
+  Only applicable for HTTP-based providers.
+     */
+    headers?: Record<string, string | undefined>;
+  }): PromiseLike<{
+    /**
+Generated images as base64 encoded strings.
```

Comment:
Don't we want some representation of what the binary format is, or to mention briefly in a doc comment if there is only one currently and no field specifier is needed?

Reply:
if you mean the image type, we can auto-detect this later using magic bytes if needed